sourcegraph/sourcegraph
# docs: On dark theme, set theme class immediately to avoid flashing

Fix for #41963. In dark mode the docs theme flashes from light to dark while loading. It happens because the site loads with the light theme as default, but then it switches to the dark theme (by setting `.theme-dark` on `body`) once `docsite.js` loads and `DOMContentLoaded` is triggered.

This proposed fix adds a short snippet in the HTML template that executes immediately and sets the dark theme class on `body` before the page loads.

## Test plan

- Set dark theme in the docs theme switcher. Load a new page to check that the dark theme loads without flashing.
- Set dark theme in OS. Load a new docs page to check that the dark theme loads without flashing.
- Set light theme or auto theme. Load a new page to check that the light theme correctly.
